### PR TITLE
Improve error message for a invalid member declaration

### LINKED ISSUE #920: Confusing error message for invalid member declaration
When the name of a `member` is incorrectly enclosed in parentheses, the error message is confusing.

For example, this code:

```f#
type T =
    static member (x) = 0
```

Causes the error:

> Only simple variable patterns can be bound in 'let rec' constructs

This is confusing, because the code above doesn't contain any `let rec` constructs.

This also happens when a type annotation is included in the parentheses, e.g.:

```f#
static member (x:int) = 0
```

I think a simple way to fix this would be to change the error message in this case to:

> Only simple variable patterns can be bound in 'member' constructs

I noticed this when looking at [this Stack Overflow question](http://stackoverflow.com/q/35101098/41071).

### PR DESCRIPTION
**Pull request by omppye**
This PR changes the error message for #920
Closes #920

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=12342) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=12342) before we can accept your contribution.<br/><br/>:x: omppye [sign now](https://cla.dotnetfoundation.org/dotnet/fsharp?pullRequest=12342)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/fsharp?pullRequest=12342) it.</sub>

**Comment by smoothdeveloper** [comment]
@omppye, that's great, could you add a test in `tests\fsharp\test.fs`, it should call `singleNegTest` function, compiling a F# code file that exhibits the error message you are addressing.

This way, the new message will be under test, as right now, it seems the CI doesn't break on this change.

**Comment by omppye** [comment]
The CI is failing because of #12348

File: src/fsharp/FSComp.txt [hunk 1]
```diff
@@ -1632,3 +1632,4 @@ reprStateMachineInvalidForm,"The state machine has an unexpected form"
 3518,implMissingInlineIfLambda,"The 'InlineIfLambda' attribute is present in the signature but not the implementation."
 3519,tcInlineIfLambdaUsedOnNonInlineFunctionOrMethod,"The 'InlineIfLambda' attribute may only be used on parameters of inlined functions of methods whose type is a function or F# delegate type."
 3520,invalidXmlDocPosition,"XML comment is not placed on a valid language element."
+3521,tcOnlySimplePatternsInMemberConstructs,"Only simple variable patterns can be bound in 'member' constructs"
```
Comment (dsyme) [inline src/fsharp/FSComp.txt:1635]: It feels like the diagnostic is unclear - beginners won't understand this.  Perhaps  `Invalid member declaration. The name of the member is missing.`

Comment (omppye) [reply src/fsharp/FSComp.txt:1635]: I think still unclear
`The name of the member is missing` is not actually true

What about `Invalid member declaration. The name of the member shouldn't be parenthesized`?

Comment (dsyme) [reply src/fsharp/FSComp.txt:1635]: The problem is that we don't know that the thing in the parentheses is the name.  I guess `Invalid member declaration. The name of the member is missing or has parentheses.`

Comment (omppye) [reply src/fsharp/FSComp.txt:1635]: Got it

**Review by dsyme — APPROVED** [review]

**Comment by dsyme** [comment]
Thanks!

**Pull request merged**